### PR TITLE
[DS-4126] 5x: Optimize docker builds

### PR DESCRIPTION
**Pull request by terrywbrady**
Port of #2307 

Images have been built on DockerHub at terrywbrady/dspace.

image

[verifyImages.sh](https://github.com/DSpace-Labs/DSpace-Docker-Images/blob/8494b5789e16cfb2f2023fc136824e38a49d4b79/tools/verifyImages.sh) has been run.

```
 ===== dspace-5_x-jdk8 =====
DSpace version:  5.11-SNAPSHOT
           JRE:  Oracle Corporation version 1.8.0_181
  ---> http://localhost:8080/xmlui
  ---> http://localhost:8080/oai/request
  ---> http://localhost:8080/solr
  ---> http://localhost:8080/rest
 ===== dspace-5_x-jdk7 =====
DSpace version:  5.11-SNAPSHOT
           JRE:  Oracle Corporation version 1.7.0_181
  ---> http://localhost:8080/xmlui
  ---> http://localhost:8080/oai/request
  ---> http://localhost:8080/solr
  ---> http://localhost:8080/rest
 ===== dspace-5_x-jdk8-test =====
DSpace version:  5.11-SNAPSHOT
           JRE:  Oracle Corporation version 1.8.0_181
  ---> http://localhost:8080/xmlui
  ---> http://localhost:8080/oai/request
  ---> http://localhost:8080/solr
  ---> http://localhost:8080/rest
 ===== dspace-5_x-jdk7-test =====
DSpace version:  5.11-SNAPSHOT
           JRE:  Oracle Corporation version 1.7.0_181
  ---> http://localhost:8080/xmlui
  ---> http://localhost:8080/oai/request
  ---> http://localhost:8080/solr
  ---> http://localhost:8080/rest
```

### CODE REVIEW

File: Dockerfile.jdk7-test [hunk 1]
```diff
@@ -1,39 +1,38 @@
 # This image will be published as dspace/dspace
 # See https://dspace-labs.github.io/DSpace-Docker-Images/ for usage details
 # 
-# This version is JDK7 compatible
+# This version is JDK8 compatible
```
Comment (PTrottier) [inline Dockerfile.jdk7-test]: See previous

File: Dockerfile.jdk7 [hunk 2]
```diff
@@ -1,34 +1,38 @@
 # This image will be published as dspace/dspace
 # See https://dspace-labs.github.io/DSpace-Docker-Images/ for usage details
 # 
-# This version is JDK7 compatible
+# This version is JDK8 compatible
```
Comment (PTrottier) [inline Dockerfile.jdk7]: I was expect this to be 

> This version is JDK7 compatible

 because this is the `jdk7` Dockerfile.

**Review by PTrottier — COMMENTED** [review]
@terrywbrady The port looks good to me. Although, I was expecting to see the community of your dog. Is this something of immediate concern for this PR?
screenshot_2019-02-07 community list

**Comment by terrywbrady** [comment]
@PTrottier , 

Thanks for catching the typos.  I update the code.

I do not see the issue that you reported.  I created new volumes and I see that the content loaded.

image

I notice that your screenshot has Mirage rather than Mirage2.  Could you give it another test?

**Comment by PTrottier** [comment]
@terrywbrady Sure, I was testing `/xmlui`, what is the address for Mirage2?

**Comment by terrywbrady** [comment]
@PTrottier , Mirage2 is an alternate theme for xmlui.  The following line of the docker-compose file will bring up Mirage2, so it makes me wonder how you initiated the dspace container.

https://github.com/DSpace-Labs/DSpace-Docker-Images/blob/master/docker-compose-files/dspace-compose/d5.override.yml#L7

If you did not use the docker-compose.yml + d5.override.yml file on startup, then the content will not autoload.

**Comment by PTrottier** [comment]
@terrywbrady I'll investigate things on my end in more detail and report back next week.

**Comment by PTrottier** [comment]
@terrywbrady Sorry, this works well for me, I can see Mirage2 as well.
screenshot_2019-02-11 dspace home

**Review by PTrottier — APPROVED** [review]

**Review by pnbecker — APPROVED** [review]
:+1:
I already reviewed the original PR and this looks similar and great to me.

**Pull request merged**